### PR TITLE
Draft10 preprations

### PR DESCRIPTION
**Pull request by simo5**
Corrections for IESG

### CODE REVIEW

**Comment by simo5** [comment]
@tomato42 @kaduk PTAL

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 1]
```diff
@@ -364,7 +363,7 @@
         </figure>
 
         <t>This value is called the exchange hash, and it is used to
-        authenticate the key exchange. The exchange hash SHOULD be kept
+        authenticate the key exchange. The exchange hash MUST be kept
```
Comment (tomato42) [inline draft-ietf-curdle-gss-keyex-sha2.xml]: I'm not so sure about this change...
RSA signature of this hash would expose it to an attacker
and it is defined as such (SHOULD, not MUST) in the original SSH RFC...

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 2]
```diff
@@ -172,9 +171,9 @@
       Elliptic Curve Cryptography are introduced. We reuse much of section 4
       of <xref target="RFC5656"/>
       to define GSS-API-authenticated ECDH Key Exchanges.</t>
-      <t>Additionally we utilize also the curves defined in <xref
-      target="I-D.ietf-curdle-ssh-curves"/> to complement the 3 classic
-      NIST defined curves required by <xref target="RFC5656"/>.</t>
+      <t>Additionally, we also utilize the curves defined in <xref
```
Comment (kaduk) [inline draft-ietf-curdle-gss-keyex-sha2.xml:175]: "Additionally" and "also" are probably redundant here (though I've seen plenty of writing that uses both like this, so it's not exactly wrong...)

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 3]
```diff
@@ -185,7 +184,7 @@
 
         <t>Key-agreement schemes ECDHE-Curve25519 and ECDHE-Curve448 perform
         the Diffie-Helman protocol using the functions X25519 and X448,
-        respectively. Implementations SHOULD compute these functions using
+        respectively. Implementations MUST compute these functions using
```
Comment (kaduk) [inline draft-ietf-curdle-gss-keyex-sha2.xml:188]: I could see some justification for not making this change in that the mathematical result could be obtained by following a different procedure than the indicated one, but being specific here is also probably fine.

File: draft-ietf-curdle-gss-keyex-sha2.xml (hunk 1 above)
Comment (simo5) [reply draft-ietf-curdle-gss-keyex-sha2.xml]: Fixed

**Review by kaduk — APPROVED** [review]
I would hold off on posting the -10 until we hear back on a couple of the active email threads.

File: draft-ietf-curdle-gss-keyex-sha2.xml (hunk 2 above)
Comment (simo5) [reply draft-ietf-curdle-gss-keyex-sha2.xml:175]: Language explicitly suggest by IETF comments :-) I just adopted it.

File: draft-ietf-curdle-gss-keyex-sha2.xml (hunk 3 above)
Comment (simo5) [reply draft-ietf-curdle-gss-keyex-sha2.xml:188]: I think if you are using a mathematically equivalent process and you are interoperable you are ok anyway, and compliant with the spirit of the RFC if not the letter. On the other hand "SHOULD" leaves open the ability for someone to say they are conformant when using something different that is not actually interoperable.
This is why I agreed changing it.

**Comment by simo5** [comment]
> I would hold off on posting the -10 until we hear back on a couple of the active email threads.

Yes, waiting on that and piling changes on this PR in the meanwhile.

File: draft-ietf-curdle-gss-keyex-sha2.xml [hunk 4]
```diff
@@ -88,20 +89,22 @@
   <middle>
     <section title="Introduction">
       <t><xref target="RFC4462">SSH GSS-API Methods</xref> allows the use of
-      GSSAPI for authentication and key exchange in SSH. It defines three
-      exchange methods all based on DH groups and SHA-1. This document
-      updates RFC4462 with new methods intended to support environments that
-      desire to use the SHA-2 cryptographic hash functions.</t>
+      <xref target="RFC2743">GSSAPI</xref> for authentication and key exchange
+      in SSH. It defines three exchange methods all based on DH groups and
+      SHA-1. This document updates RFC4462 with new methods intended to support
```
Comment (tomato42) [inline draft-ietf-curdle-gss-keyex-sha2.xml:94]: shouldn't RFC4462 be in xref here?

**Review by tomato42 — APPROVED** [review]
one nit, fine with handling it either way

File: draft-ietf-curdle-gss-keyex-sha2.xml (hunk 4 above)
Comment (simo5) [reply draft-ietf-curdle-gss-keyex-sha2.xml:94]: nobody request it so I am ok leaving as is, it's not like links to RFC4462 are not plentiful in the document already :-)

**Comment by simo5** [comment]
Ok I think we waited enough for input and none came, cutting draft 10 and submitting

**Pull request merged**